packages/cloud_firestore/cloud_firestore/android/src/main/java/io/flutter/plugins/firebase/cloudfirestore/CloudFirestorePlugin.java

```diff
@@ -55,11 +59,12 @@
 import java.util.Map;
 import java.util.concurrent.TimeUnit;
 
-public class CloudFirestorePlugin implements MethodCallHandler {
+public class CloudFirestorePlugin implements MethodCallHandler, FlutterPlugin, ActivityAware {
 
   private static final String TAG = "CloudFirestorePlugin";
-  private final MethodChannel channel;
-  private final Activity activity;
+  private MethodChannel channel;
+  private Activity activity;
+  private ActivityPluginBinding activityPluginBinding;
 
```

Comment:
You don't need to keep this binding to get activity lazily. When attaching to activity, get activity from binding is good enough. V2 embedder will make sure it's available unlike registrar.